internal/stage_write_tree.go

```diff
@@ -80,7 +80,7 @@ func testWriteTree(stageHarness *tester_utils.StageHarness) error {
 	logger.Debugf("Reading file at %v", relativePath)
 
 	gitObjectFileContents, err := os.ReadFile(gitObjectFilePath)
-	if err == os.ErrNotExist {
+	if errors.Is(err, os.ErrNotExist) {
```

Comment:
@libmartinito can we add tests for this? Seems like something that's easy to get wrong again